### DISCUSSION TITLE
Should slint-viewer be interactable?

### DISCUSSION THREAD

**Discussion by preland** [slint-ui/slint#4110] [Q&A]
I don't want to post an issue unless something is genuinely wrong. I am currently trying to use slint-viewer on the default slint project for Rust (the button and Counter one). Running it shows the project correctly (although it throws two cryptic "QGraphicsEffectSource::draw: Can only begin as a result of QGraphicsEffect::draw" errors on startup. These errors occur when normally running the application, and occur whenever the UI is updated in any way). However, the Counter button does not increase the counter value, although the button is still animated, so the program isn't completely hanging as far as I can tell. 

To further complicate things, running with the --auto-reload option has all the above issues as well, but editing the .slint file throws a "error: Could not load appwindow.slint: No such file or directory (os error 2), which I presume _does_ cause the program to hang, as further edits do not result in further errors.

I also tried setting the -I and -L directories to the src/ folder of the project, which (unsuprisingly to me at least) didn't change anything.

I don't see anywhere in the documentation for slint-viewer it mentioning whether it can or can't support interactions as described. I'd normally be okay with accepting that realtime interaction isn't a feature (like most other UI preview tools), but the --auto-reload feature not working throws me off.

Any help is appreciated (Platform: Intel MacOS ; slint: 1.3.2; slint-viewer: 1.3.2)

**Comment by hunger** [comment]
First of: Thanks for your time! It is effort to report issues and that is very much appreciated. It is totally fine to open bugs: We can just close them or move them to a more appropriate place when needed.

Slint-viewer does only run the .slint code. So none of the callbacks into your business logic is ever triggered from the viewer. This can lead to counters not updating and similar issues. So I am not too surprised about that part.

Could you try pointing the Slint viewer to the absolute path of your appwindow.slint file? If that works, then slint-viewer is probably running in some directory you were not expecting it to run in. Resolving the relative path to appwindow.slint does just not work then and without the path actually resolving to some file, setting up a file watcher will not work either.
